addons/hr_holidays/models/hr_employee.py

```diff
@@ -0,0 +1,107 @@
+    _inherit = "hr.employee"
+
+    remaining_leaves = fields.Float(compute='_compute_remaining_days', string='Remaining Legal Leaves', inverse='_inverse_remaining_days', help='Total number of legal leaves allocated to this employee, change this value to create allocation/leave request. Total based on all the leave types without overriding limit.')
+    current_leave_state = fields.Selection([('draft', 'New'), ('confirm', 'Waiting Approval'),
+        ('refuse', 'Refused'), ('validate1', 'Waiting Second Approval'),
+        ('validate', 'Approved'), ('cancel', 'Cancelled')],
+        compute='_compute_leave_status', string="Current Leave Status")
+    current_leave_id = fields.Many2one('hr.holidays.status', compute='_compute_leave_status', string="Current Leave Type")
+    leave_date_from = fields.Date(compute='_compute_leave_status', string='From Date')
+    leave_date_to = fields.Date(compute='_compute_leave_status', string='To Date')
+    leaves_count = fields.Integer(compute='_compute_leaves_count', string='Number of Leaves')
+    show_leaves = fields.Boolean(compute='_compute_show_leaves', string="Able to see Remaining Leaves")
+    is_absent_totay = fields.Boolean(compute='_compute_absent_employee', search='_search_absent_employee', string="Absent Today")
+
+    def _remaining_leaves_count(self):
+        leaves = self.env['hr.holidays'].read_group([('state', '=', 'validate'),
+            ('holiday_status_id.limit', '=', False), ('employee_id', 'in', self.ids)],
+            ['number_of_days', 'employee_id'], ['employee_id'])
+        leaves = dict([(leave['employee_id'][0], leave['number_of_days']) for leave in leaves])
+        return dict([(employee.id, leaves.get(employee.id, 0)) for employee in self])
+
+    def _inverse_remaining_days(self):
+        if self.remaining_leaves:
+            diff = self.remaining_leaves - self._remaining_leaves_count().get(self.id)
+            # Find for holidays status
+            status = self.env['hr.holidays.status'].search([('limit', '=', False)])
+            if len(status) != 1:
+                raise UserError(_("The feature behind the field 'Remaining Legal Leaves' can only be used when there is only one leave type with the option 'Allow to Override Limit' unchecked. (%s Found). Otherwise, the update is ambiguous as we cannot decide on which leave type the update has to be done. \nYou may prefer to use the classic menus 'Leave Requests' and 'Allocation Requests' located in 'Human Resources \ Leaves' to manage the leave days of the employees if the configuration does not allow to use this field.") % (len(status)))
+            if diff > 0:
+                leave = self.env['hr.holidays'].create({'name': _('Allocation for %s') % self.name, 'employee_id': self.id, 'holiday_status_id': status.id, 'type': 'add', 'holiday_type': 'employee', 'number_of_days_temp': diff})
+            elif diff < 0:
+                raise UserError(_('You cannot reduce validated allocation requests'))
```

Comment:
The code would be simpler if it started with all error checks, and only then followed with "normal" code.